### PR TITLE
Update md5 Deprecation

### PR DESCRIPTION
**Pull request by harsitagarwalla187**
Pull Request for Issue #39701.

### Summary of Changes
Updated md5() from passing null to parameter.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by laoneo** [comment]
The second argument of md5 is a boolean and not a string. Why do you add a string here?

File: libraries/src/Cache/CacheStorage.php [hunk 1]
```diff
@@ -97,8 +97,8 @@ public function __construct($options = [])
     {
         $app = Factory::getApplication();
 
-        $this->_hash        = md5($app->get('secret'));
-        $this->_application = $options['application'] ?? md5(JPATH_CONFIGURATION);
+        $this->_hash        = md5($app->get('secret'), '');
```
Comment (alikon) [inline libraries/src/Cache/CacheStorage.php]: should be something like
`md5($app->get('secret', ''));`

File: libraries/src/Cache/CacheStorage.php [hunk 2]
```diff
@@ -97,8 +97,8 @@ public function __construct($options = [])
...
+        $this->_application = $options['application'] ?? md5(JPATH_CONFIGURATION, '');
```
Comment (alikon) [inline libraries/src/Cache/CacheStorage.php]: this is wrong

File: libraries/src/Cache/CacheStorage.php [hunk 3]
```diff
@@ -360,7 +360,7 @@ public function unlock($id, $group = null)
      */
     protected function _getCacheId($id, $group)
     {
-        $name          = md5($this->_application . '-' . $id . '-' . $this->_language);
+        $name          = md5($this->_application . '-' . $id . '-' . $this->_language, '');
```
Comment (alikon) [inline libraries/src/Cache/CacheStorage.php]: this is wrong

**Review by alikon — CHANGES_REQUESTED** [review]

**Comment by harsitagarwalla187** [comment]
@alikon Please have a look. I have made the changes.

**Review by laoneo — APPROVED** [review]

**Review by alikon — APPROVED** [review]

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 633cc2179e9a48353a550b78c7cd8f22551b94e6<br /><br />Tested with null as secret in configuration. Not sure how this can happen, but the fix works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39720">issues.joomla.org/tracker/joomla-cms/39720</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 633cc2179e9a48353a550b78c7cd8f22551b94e6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39720">issues.joomla.org/tracker/joomla-cms/39720</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39720">issues.joomla.org/tracker/joomla-cms/39720</a>.</sub>

**Comment by fancyFranci** [comment]
Thank you!

**Pull request merged**